backend/capellacollab/projects/toolmodels/backups/runs/crud.py

```diff
@@ -52,6 +54,17 @@ def get_scheduled_and_running_pipelines(
     )
 
 
+def get_pipeline_runs_for_pipeline(
+    db: Session, pipeline: pipeline_models.DatabaseBackup
+):
+    return fastapi_pagination.ext.sqlalchemy.paginate(
+        db,
+        select(models.DatabasePipelineRun)
+        .where(models.DatabasePipelineRun.pipeline == pipeline)
+        .order_by(models.DatabasePipelineRun.id.desc()),
+    )
```

Comment:
First, I would not provide the whole pipeline here but rather just the `pipeline_id`. So then having,
`def get_pipeline_runs_by_pipeline_id(db: Session, pipeline_id: int)`.

In addition, IMO we should not return the`fastapi_pagination.ext.sqlalchemy.paginate(...)` type from within our `crud.py` like this. I have a few suggestions how we could do it:
1. We could keep the pagination functions in our `curd.py` and rename the function name to `get_pipeline_runs_for_pipeline_id_paginated`
2. We could create a separate `pagination.py` where we then put this function in without any name changes.
3. Do not use the sqlalchemy extension but rather put the resulting list/sequence from the select in the "normal" pagination function

Reply:
> First, I would not provide the whole pipeline here but rather just the `pipeline_id`. So then having, `def get_pipeline_runs_by_pipeline_id(db: Session, pipeline_id: int)`.

https://github.com/DSD-DBS/capella-collab-manager/pull/717/files#diff-7702246c9b82e90105dd8e55cd049dcafd3f6db16587ecf3909f19c85c5a00a7

> In addition, IMO we should not return the`fastapi_pagination.ext.sqlalchemy.paginate(...)` type from within our `crud.py` like this. I have a few suggestions how we could do it:
> 
> 1. We could keep the pagination functions in our `curd.py` and rename the function name to `get_pipeline_runs_for_pipeline_id_paginated`

This sounds like the best solution for me. 

> 2. We could create a separate `pagination.py` where we then put this function in without any name changes.

As we have only one paginated function, this doesn't really make sense yet. And it belongs to the crud module, I'd say.


> 3. Do not use the sqlalchemy extension but rather put the resulting list/sequence from the select in the "normal" pagination function

This is very inefficient, esp. with a growing number of pipeline runs.

Reply:
Then let's use the first version and simply rename the function. It might be useful to include this in our development documentation, probably in the style guide, so that we use this scheme consistently when we use pagination.